Submitted By: Alexander E. Patrakov (semzx at newmail dot ru)
Origin: FreeType 2 CVS
Date: 2003-11-02
Initial Package Version: 2.1.5
Description: From FreeType 2 changelog:
2003-10-19  Werner Lemberg  <wl@gnu.org>

        * src/autohint/ahhint.c (ah_hinter_load_glyph): Patch from
        2003-08-18 introduced a severe bug (FT_Render_Glyph was called
        twice under some circumstances, causing strange results).  This
        is fixed now by clearing the FT_LOAD_RENDER bit of `load_flags'.

These strange results confuse e.g. DirectFB (just nothing is drawn with
TrueType fonts). Please notify me if you find other victims.

===================================================================
RCS file: /home/freetype/cvs/cvs/freetype/freetype2/src/autohint/ahhint.c,v
retrieving revision 1.71
retrieving revision 1.72
diff -u -r1.71 -r1.72
--- freetype2/src/autohint/ahhint.c	2003/08/17 23:00:59	1.71
+++ freetype2/src/autohint/ahhint.c	2003/10/20 08:29:45	1.72
@@ -1850,9 +1850,9 @@
 
     hinter->do_stem_adjust   = FT_BOOL( hint_mode != FT_RENDER_MODE_LIGHT );
 
-
-    load_flags  |= FT_LOAD_NO_SCALE
-                 | FT_LOAD_IGNORE_TRANSFORM ;
+    load_flags |= FT_LOAD_NO_SCALE
+                | FT_LOAD_IGNORE_TRANSFORM;
+    load_flags &= ~FT_LOAD_RENDER;
 
     error = ah_hinter_load( hinter, glyph_index, load_flags, 0 );
 
